lib/dm-core/property/typecast/numeric.rb

```diff
@@ -15,7 +15,7 @@ module Numeric
         # @api private
         def typecast_to_numeric(value, method)
           if value.respond_to?(:to_str)
-            if value.to_str =~ /\A(-?(?:0|[1-9]\d*)(?:\.\d+)?|(?:\.\d+))\z/
+            if value.to_str =~ /\A(-?(?:[0-9]\d*)(?:\.\d+)?|(?:\.\d+))\z/
```

Comment:
Yes it is. The original syntax was to prevent nonsense like `0000000001`, but it turns out there are lots of valid cases where a leading number is used, eg: `0.99`.

I agree with the approach taken in virtus (which I actually merged a couple of years ago), where we take anything that is a number and coerce it. This is with the understanding that the developer may (and probably should) further constrain the input to some valid range rather than allowing any number.